lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,404 @@
+                                                       supports_check_mode=True,
+                                                       supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
+        if not self.location:
+            self.location = resource_group.location
+
+        # Check if the ACI instance already present in the RG
+        if self.state == 'present':
+
+            self.log("Need to Create / Update the ACI instance")
+
+            if self.check_mode:
+                return self.results
+
+            self.results['state'] = self.create_update_aci()
+            self.results['changed'] = True
+
+            self.log("Creation / Update done")
+        elif self.state == 'absent':
+            self.delete_aci()
+            self.log("ACI instance deleted")
+
+        return self.results
+
+    def create_update_aci(self):
+        '''
+        Creates or updates a container service with the specified configuration of orchestrator, masters, and agents.
+
+        :return: deserialized ACI instance state dictionary
+        '''
+        self.log("Creating / Updating the ACI instance {0}".format(self.name))
+
+        # self.log("orchestrator_profile : {0}".format(parameters.orchestrator_profile))
+        # self.log("service_principal_profile : {0}".format(parameters.service_principal_profile))
+        # self.log("linux_profile : {0}".format(parameters.linux_profile))
+        # self.log("ssh from yaml : {0}".format(results.get('linux_profile')[0]))
+        # self.log("ssh : {0}".format(parameters.linux_profile.ssh))
+        # self.log("master_profile : {0}".format(parameters.master_profile))
+        # self.log("agent_pool_profiles : {0}".format(parameters.agent_pool_profiles))
+        # self.log("vm_diagnostics : {0}".format(parameters.diagnostics_profile.vm_diagnostics))
+
+        credentials = None
+
+        if self.registry_login_server is not None:
+            credentials = ImageRegistryCredential(self.registry_login_server,
+                                                  self.registry_username,
+                                                  self.registry_password)
+
+        ip_address = None
+
+        if self.ip_address is not None:
+                ip_address = IpAddress([self.port], self.ip_address)
+
+        parameters = ContainerGroup(self.location,
+                                    None,
+                                    [Container(self.name, self.image, ResourceRequirements(ResourceRequests(self.memory, self.cpu)))],
```

Comment:
I think we need to enable user to create multi-container group like documentation [here](https://docs.microsoft.com/en-us/azure/container-instances/container-instances-multi-container-group).

Reply:
agree

Reply:
that's in the second step. i am worried it's going to get very messy.
so i want to have full single container scenario completed before doing second step.

Reply:
Sounds good!
You may want to pay more attention to your parameters for multi-container group support. We should avoid breaking interfaces once they are released.